lib/rules/callback-return.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @fileoverview Enforce return after a callback.
+ * @author Jamund Ferguson
+ * @copyright 2015 Jamund Ferguson. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    var callbacks = context.options[0] || ["callback", "cb", "next"];
+
+    //--------------------------------------------------------------------------
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Find the closest parent matching a list of types.
+     * @param {ASTNode} node The node whose parents we are searching
+     * @param {Array} types The node types to match
+     * @returns {ASTNode} The matched node or undefined.
+     */
+    function findClosestParentOfType(node, types) {
+        if (!node.parent) {
+            return null;
+        }
+        if (types.indexOf(node.parent.type) === -1) {
+            return findClosestParentOfType(node.parent, types);
+        }
+        return node.parent;
+    }
+
+    /**
+     * Check to see if a CallExpression is in our callback list.
+     * @param {ASTNode} node The node to check against our callback names list.
+     * @returns {Boolean} Whether or not this function matches our callback name.
+     */
+    function isCallback(node) {
+        return node.callee.type === "Identifier" && callbacks.indexOf(node.callee.name) > -1;
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    return {
+        "CallExpression": function(node) {
+            var parent, nextNodes, keywords;
+
+            if (isCallback(node)) {
+                parent = findClosestParentOfType(node, ["ReturnStatement", "FunctionExpression", "FunctionDeclaration", "IfStatement"]);
+            }
+
+            if (!parent) {
+                return;
+            }
+
+            // if you we are in an if statement we need to ensure we return soon
+            if (parent.type === "IfStatement") {
+
+                // try to figure out the next thing that happens
+                nextNodes = context.getTokensAfter(node, 10); // punctuation, comment, else, return, whatever
```

Comment:
haha yes of course it will fail. looking for suggestions on another way. currently messing with some of the code from `no-unreachable`  to find something more robust. so far no luck, but not giving up.